### PR TITLE
add documentation site as new workspace

### PR DESCRIPTION
**Pull request by Rishav-12**
What is this PR about? (docs, fix, feature, style, devops, etc.)
- docs

Related issue number: #1 

Description of changes:
- add a new documentation website inside the `docs` directory
- add a home page and a page about the Reactive Form

Tag a reviewer: @ayoayco 

Tasks:
- [x] I have ran the tests to make sure nothing is broken (see CONTRIBUTING.md)
- [x] I have mentioned the issue this PR is addressing

Note:
- I kept the default layout of the site
- The package.json inside the `docs` folder should be changed if needed as it doesn't link to the correct GitHub url
- **[Important]** Please review the changes carefully as there may be mistakes since I'm absolutely new to astro

### CODE REVIEW

**Comment by ayoayco** [comment]
Thanks for the PR @Rishav-12! 🚀 Let's try to get this in. :)

First we need to make the tests pass. I think you need to get the package-lock.json from the main branch and then do "npm i".

**Comment by Rishav-12** [comment]
On it

File: docs/src/components/Footer/AvatarList.astro [hunk 1]
```diff
@@ -0,0 +1,167 @@
+---
+// fetch all commits for just this page's path
+type Props = {
+	path: string;
+};
+const { path } = Astro.props as Props;
+const resolvedPath = `examples/docs/${path}`;
+const url = `https://api.github.com/repos/withastro/astro/commits?path=${resolvedPath}`;
+const commitsURL = `https://github.com/withastro/astro/commits/main/${resolvedPath}`;
```
Comment (ayoayco) [inline docs/src/components/Footer/AvatarList.astro:9]:
```suggestion
const url = `https://api.github.com/repos/ayoayco/astro-reactive-library/commits?path=${resolvedPath}`;
const commitsURL = `https://github.com/ayoayco/astro-reactive-library/commits/main/${resolvedPath}`;
```

File: docs/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+export const SITE = {
+	title: 'Documentation',
+	description: 'Your website description.',
+	defaultLanguage: 'en_US',
```
Comment (ayoayco) [inline docs/src/config.ts:4]:
```suggestion
	title: 'Astro Reactive Library Docs',
	description: 'Documentation for the Astro Reactive Library',
	defaultLanguage: 'en_US',
```

File: docs/src/config.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	ogLocale?: string;
+	lang?: string;
+};
+
+export const KNOWN_LANGUAGES = {
+	English: 'en',
+} as const;
+export const KNOWN_LANGUAGE_CODES = Object.values(KNOWN_LANGUAGES);
+
+export const GITHUB_EDIT_URL = `https://github.com/withastro/astro/tree/main/examples/docs`;
```
Comment (ayoayco) [inline docs/src/config.ts:33]:
```suggestion
export const GITHUB_EDIT_URL = `https://github.com/ayoayco/astro-reactive-library/tree/main/docs`;
```

**Review by ayoayco — CHANGES_REQUESTED** [review]
Great work! I have some suggestions to update. Please let me know if they work :)

**Review by ayoayco — APPROVED** [review]
Thanks for the great update! Now the project has a docs website that we can improve more! :)

**Pull request merged**